app/models/ability.rb

```diff
@@ -0,0 +1,21 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the passed in user here. For example:
+    #
+    #   user ||= User.new # guest user (not logged in)
+    #   if user.admin?
+    #     can :manage, :all
+    #   else
+    #     can :read, :all
+    #   end
+
+    can :destroy, Post, author_id: user.id
+    can :destroy, Comment, user_id: user.id
+    return unless user.role == 'Administrator'
+
+    can :destroy, Post
+    can :destroy, Comment
```

Comment:
- [ ] Instead of using `destroy`  you should use `manage` so the admin or owner of the post/comment can perform all view, update, or destroy operations and also authorize other users to read all the entities. Please make sure to fix this issue.
  - You can consider using this code.
  
```ruby
 def initialize(user)

         user ||= User.new # guest user (not logged in)

         if user.role == 'Administrator'
         can :manage, :all
         else
         can :manage, Post, author: user 
         can :manage, Comment, author: user
         can :read, :all
         end
     end
```